mon-pix/app/services/session.js

```diff
@@ -0,0 +1,147 @@
+import { inject as service } from '@ember/service';
+import SessionService from 'ember-simple-auth/services/session';
+import get from 'lodash/get';
+
+export default class CurrentSessionService extends SessionService {
+  @service currentUser;
+  @service currentDomain;
+  @service intl;
+  @service moment;
+  @service url;
+  @service router;
+
+  routeAfterAuthentication = 'user-dashboard';
+
+  async handleAuthentication() {
+    await this._loadCurrentUserAndSetLocale();
+
+    const nextURL = this.data.nextURL;
```

Comment:
Dans location.js on avait cette méthode privée 
`  _isFromIdentityProviderLoginPage(nextUrl, identityProviderCode) {
    const isUserLoggedInToIdentityProvider =
      get(this.session, 'data.authenticated.identity_provider_code') === identityProviderCode;
    return nextUrl && isUserLoggedInToIdentityProvider;
  }`
  
  appellée dans 
`  const nextURL = this.session.data.nextURL;
    const isFromIdentityProviderLoginPage = Object.keys(IdentityProviders).some((key) =>
      this._isFromIdentityProviderLoginPage(nextURL, IdentityProviders[key].code)
    );
    if (isFromIdentityProviderLoginPage) {
      this.session.set('data.nextURL', undefined);
      this.router.replaceWith(nextURL);
    } else {
      _super.call(this, ...arguments);
    }`